dynamic_rest/serializers.py

```diff
@@ -213,11 +213,20 @@ def __init__(
         super(WithDynamicSerializerMixin, self).__init__(**kwargs)
 
         self.envelope = envelope
-        self.sideloading = sideloading
         self.debug = debug
         self.dynamic = dynamic
         self.request_fields = request_fields or {}
-        self.embed = embed if sideloading is None else not sideloading
+
+        # sideloading modifies top-level response keys,
+        # so it requires an envelope
+        if envelope is False:
```

Comment:
Might be good to check for conflicting params, like `envelope == False and sideloading == True` and raise an error.